File: 1.0/src/c/cz/mg/vulkan/PFN_vkDestroyRenderPass.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyRenderPass_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong renderPass, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyRenderPass f = (PFN_vkDestroyRenderPass)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkRenderPass*)jniLongToPointer(renderPass)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkDestroyRenderPass.java
package cz.mg.vulkan;

public class PFNvkDestroyRenderPass extends VkFunctionPointer {
    public PFNvkDestroyRenderPass() {
    }

    protected PFNvkDestroyRenderPass(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkDestroyRenderPass(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyRenderPass(long value) {
        setValue(value);
    }

    public PFNvkDestroyRenderPass(VkInstance instance) {
        super(instance, new VkString("vkDestroyRenderPass"));
    }

    public void call(VkDevice device, VkRenderPass renderPass, VkAllocationCallbacks pAllocator){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), renderPass != null ? renderPass.getVkAddress() : VkPointer.getNullAddressNative(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long renderPass, long pAllocator);

}
